From e2144cbd1b7f71c2fc57264f125ef298c60f22b7 Mon Sep 17 00:00:00 2001
From: Kevin Cernekee <cernekee@gmail.com>
Date: Mon, 30 Dec 2013 17:49:37 -0800
Subject: [PATCH] auth-dialog: Add support for OC_FORM_OPT_IGNORE

Hide any options that libopenconnect doesn't want us to display.


diff --git a/auth-dialog/main.c b/auth-dialog/main.c
index 4ab28cb..2174ea8 100644
--- a/auth-dialog/main.c
+++ b/auth-dialog/main.c
@@ -75,11 +75,13 @@
 #define AUTHGROUP_OPT(form)		(void *)(form)->authgroup_opt
 #define AUTHGROUP_SELECTION(form)	(form)->authgroup_selection
 #define FORMCHOICE(sopt, i)		((sopt)->choices[i])
+#define IGNORE_OPT(opt)			((opt)->flags & OC_FORM_OPT_IGNORE)
 #else
 #define NEWGROUP_SUPPORTED		0
 #define AUTHGROUP_OPT(form)		NULL
 #define AUTHGROUP_SELECTION(form)	0
 #define FORMCHOICE(sopt, i)		(&(sopt)->choices[i])
+#define IGNORE_OPT(opt)			0
 #define OC_FORM_RESULT_ERR		-1
 #define OC_FORM_RESULT_OK		0
 #define OC_FORM_RESULT_CANCELLED	1
@@ -704,7 +706,8 @@ static gboolean ui_form (struct oc_auth_form *form)
 	for (opt = form->opts; opt; opt = opt->next) {
 		ui_fragment_data *data;
 
-		if (opt->type == OC_FORM_OPT_HIDDEN)
+		if (opt->type == OC_FORM_OPT_HIDDEN ||
+		    IGNORE_OPT(opt))
 			continue;
 
 		data = g_slice_new0 (ui_fragment_data);
--
Gitg

